### DISCUSSION TITLE
Get current zoom level reactive number

### DISCUSSION THREAD

**Discussion by gitHoussem** [bcakmakoglu/vue-flow#1632] [Q&A] [answered]
Hello, 

I am currently trying to implement my own zoom in/out buttons without relying on the default ones in the designer. 
The zooming functions were easy to use `zoomIn` `zoomOut` and `zoomTo` (and I thank you for that).
However, I am unable to get the current zoom level number and get it to be reactive so that my input that has the zoom percentage updates automatically. (users can also zoom using the trackpad by swiping 2 fingers, so It's not something that can be hardcoded in the onClick of the zoom buttons).

![image](https://github.com/user-attachments/assets/80953f99-9266-4870-8278-99220380cd44)

<img width="268" alt="image" src="https://github.com/user-attachments/assets/a12ad6d6-ba4c-4544-af7c-cbfac8c844e1">

Right now, here is how I am getting the zoom level 

```ts
import { useVueFlow } from '@vue-flow/core';

const { getViewport } = useVueFlow();
const viewport = getViewport();
console.log(viewport.zoom);  // This will print the current zoom level
```
My question is : How can I get a reactive zoom level value.

Thank you !

**Comment by bcakmakoglu** [comment, marked as answer]
Instead of `getViewport` you can just use 

```ts
const { viewport } = useVueFlow()

watch(() => viewport.value.zoom, (nextZoom) => {
	console.log(nextZoom)
})
```

**Reply by gitHoussem** [reply]
You are the best ! Thank you
